tests/functional/directives.ts

```diff
@@ -0,0 +1,655 @@
+          return "queryWithUpper";
+        }
+
+        @Query()
+        @Directive("@upper")
+        queryWithUpperSDL(): string {
+          return "queryWithUpper";
+        }
+
+        @Query()
+        @Directive("append")
+        queryWithAppend(): string {
+          return "hello";
+        }
+
+        @Query()
+        @Directive("@append")
+        queryWithAppendSDL(): string {
+          return "hello";
+        }
+
+        @Query()
+        @Directive("append")
+        @Directive("upper")
+        queryWithUpperAndAppend(): string {
+          return "hello";
+        }
+
+        @Query()
+        @Directive("@append @upper")
+        queryWithUpperAndAppendSDL(): string {
+          return "hello";
+        }
+
+        @Mutation()
+        @Directive("foo")
+        mutationWithDirective(): string {
+          return "mutationWithDirective";
+        }
+
+        @Mutation()
+        @Directive("bar", { baz: "true" })
+        mutationWithDirectiveWithArgs(): string {
+          return "mutationWithDirectiveWithArgs";
+        }
+
+        @Mutation()
+        @Directive("upper")
+        mutationWithUpper(): string {
+          return "mutationWithUpper";
+        }
+
+        @Mutation()
+        @Directive("@upper")
+        mutationWithUpperSDL(): string {
+          return "mutationWithUpper";
+        }
+
+        @Mutation()
+        @Directive("append")
+        mutationWithAppend(): string {
+          return "hello";
+        }
+
+        @Mutation()
+        @Directive("@append")
+        mutationWithAppendSDL(): string {
+          return "hello";
+        }
+
+        @Mutation()
+        @Directive("append")
+        @Directive("upper")
+        mutationWithUpperAndAppend(): string {
+          return "hello";
+        }
+
+        @Mutation()
+        @Directive("@append @upper")
+        mutationWithUpperAndAppendSDL(): string {
+          return "hello";
+        }
+      }
+
+      schema = await buildSchema({
+        resolvers: [SampleResolver],
+      });
+
+      SchemaDirectiveVisitor.visitSchemaDirectives(schema, {
+        upper: UpperCaseDirective,
+        append: AppendDirective,
+      });
+    });
+
+    it("should generate schema without errors", async () => {
+      expect(schema).toBeDefined();
+    });
+
+    describe("Query", () => {
+      it("should add directives to query types", async () => {
+        const queryWithDirective = schema.getQueryType()!.getFields().queryWithDirective;
+
+        assertValidFieldDirective(queryWithDirective.astNode, "foo");
+      });
+
+      it("should add directives to query types with arguments", async () => {
+        const queryWithDirectiveWithArgs = schema.getQueryType()!.getFields()
+          .queryWithDirectiveWithArgs;
+
+        assertValidFieldDirective(queryWithDirectiveWithArgs.astNode, "bar", { baz: "true" });
+      });
+
+      it("calls directive 'upper'", async () => {
+        const query = `query {
+        queryWithUpper
```

Comment:
formatting issues all the way down

Reply:
It’d be amazing if this repo integrated prettier to avoid formatting issues. Would you be open to this? @19majkel94 I could work on it.

Reply:
Prettier is integrated with this repo and a commit hook. But prettier doesn't format string, it only recognise `gql` template tag.

Reply:
VSCode copy/paste kept doing this.